### ISSUE TITLE
Empty quoted strings are parsed as null

### ISSUE THREAD

**Issue by JensAyton** [jpsim/Yams#301]
_Note:_ This is not the same as #253. However, the issue @tjprescott referenced in #253 is actually this.

---

When decoding a document like this:
```yaml
stringA: "Hello"
stringB: ""
```
into a struct with optional string members like this:
```swift
struct TestType: Codable {
    var stringA: String?
    var stringB: String?
    var stringC: String?
}
```
`stringB` is read as nil rather than an empty string. If `TestType.stringB` is made non-optional, it’s read as an empty string.

On further inspection, this could also happen for string values of `"~"`, `"null"`, `"Null"` or `"NULL"`. This is surprising and [dangerous](https://www.wired.com/2015/11/null/) behaviour.

The YAML spec explicitly states: “Note that a null is different from an empty string.” While the style of the spec is confusing, I don’t believe it intends for the string `"null"` to be of null type either.

This can be fixed by changing `NSNull.construct` to return nil unless `scalar.style == .plain`. However, I’m leery of making a PR with this change since it may impact existing code with unfortunate dependencies on the current behaviour.

**Comment by JensAyton** [comment]
Full example:

```swift
import Foundation
import Yams

let testYAML = #"""
stringA: "Hello"
stringB: ""
"""#

let testJSON = #"""
{
    "stringA": "Hello",
    "stringB": ""
}
"""#.data(using: .utf8)!

struct TestType: Codable {
    var stringA: String?
    var stringB: String?
    var stringC: String?
}

// Prints wrong result TestType(stringA: Optional("Hello"), stringB: nil, stringC: nil)
let testFromYAML = try! YAMLDecoder().decode(TestType.self, from: testYAML)
print(String(reflecting: testFromYAML))

// Prints correct result TestType(stringA: Optional("Hello"), stringB: nil, stringC: nil)
let testFromJSON = try! JSONDecoder().decode(TestType.self, from: testJSON)
print(String(reflecting: testFromJSON))
```

**Comment by jpsim** [comment]
Fixed in 0c4ff784fbce79060be2722f27797447789de1a2.
